packages/database/src/browser/MigrationManager.js

```diff
@@ -0,0 +1,145 @@
+import { PGlite } from '@electric-sql/pglite';
+import path from 'path';
+import { getConnectionConfig } from './getConnectionConfig';
+
+const DEFAULT_MIGRATIONS_PATH = path.resolve(__dirname, '/packages/database/src/core/migrations');
+
+export class MigrationManager {
+  client = null;
+  config = null;
+
+  constructor(config = {}) {
+    this.config = {
+      ...config,
+      migrationsPath: config.migrationsPath || DEFAULT_MIGRATIONS_PATH,
+      logger: config.logger || {
+        info: message => console.log(message),
+        error: message => console.error(message),
+      },
+    };
+    this.client = new PGlite(getConnectionConfig);
+  }
+
+  async initialize() {
+    await this.createMigrationsTable();
+  }
+
+  async close() {
+    await this.client.close();
+  }
+
+  async createMigrationsTable() {
+    await this.client.exec(`
+      CREATE TABLE IF NOT EXISTS migrations (
+        id SERIAL PRIMARY KEY,
+        version INTEGER NOT NULL,
+        name VARCHAR(255) NOT NULL,
+        executed_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+      );
+    `);
+  }
+
+  async loadMigrationFiles() {
+    const migrationFiles = import.meta.glob('../../core/migrations/*.js', { eager: true });
+    const migrations = [];
+
+    const sortedMigrationFiles = Object.fromEntries(
+      Object.entries(migrationFiles).sort((a, b) => a[0].localeCompare(b[0])),
+    );
+
+    for (const [name, migrationModule] of Object.entries(sortedMigrationFiles)) {
+      const targets = migrationModule._meta.targets || [];
+
+      // Only run migration if it is intended for the browser
+      if (!targets.includes('browser')) {
+        continue;
+      }
```

Comment:
1 of the important points of the PR is here. I want the browser migration manager to be able to:
- Share the migration folder and run migrations created from `db-generate` (which is the server migration tool), so that we don't have to have 2 different set of migrations for datatrak-web and server
- Have the ability to run migration that only applies for datatrak-web because:
  - We do not want to run historical migrations that already exist in the migrations folder

Hence, I utilised the existing `_meta` export that is also used in db-migrate, and add an extra property targets, so it would look something like this:

```
exports._meta = {
  version: 1,
  targets: ['server', 'browser']
};
```

And browser's MigrationManager will check for browser migrations only

Reply:
I really love it!!